src/ui/widgets/EmbeddedDisplay/embeddedDisplay.tsx

```diff
@@ -40,41 +41,56 @@ export const EmbeddedDisplay = (
 
   log.debug(description);
 
-  // Apply the height to the top level if relative layout and none have been provided
-  if (props.position instanceof RelativePosition) {
-    props.position.height = props.position.height || description.height;
-    props.position.width = props.position.width || description.width;
-  }
+  // Check whether to override OPI autoZoomToFit
+  const autoZoomToFitOverride =
+    typeof props.autoZoomToFitOverride === "undefined"
+      ? true
+      : props.autoZoomToFitOverride;
 
-  // Apply the height to the top level if relative layout and none have been provided
-  if (props.position instanceof RelativePosition) {
-    props.position.height = props.position.height || description.height;
-    props.position.width = props.position.width || description.width;
-  }
+  // Get the screen height
```

Comment:
Is there the possibility to factor any of this stuff out into a function? This may be complicated as you are operating on props etc

Reply:
Yes, very good idea as I use this method of determining the value of an optional parameter multiple times. I have created a function and moved it to 'utils.ts'. I then use this in both the embeddedDisplay and display code.